From ea38e4b9a503f243ccccff91c41ccb0b4406b131 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 30 Oct 2008 17:12:02 +1100
Subject: [PATCH] initscripts: disable several useless things on boot that slow it down

1. remove echo of "System bootup..." - conflicts with splashscreens like
psplash and exquisitie. there is no need for it.
2. remove MOTD handling. it's disabled anyway and it's an embedde device.
3. disable ldconfig on boot. this should have been handled by the installer
on image build and opkg on package install. no reason to run every boot.
---
 packages/initscripts/initscripts-1.0/bootmisc.sh |   20 ++++++++++++--------
 packages/initscripts/initscripts_1.0.bb          |    2 +-
 2 files changed, 13 insertions(+), 9 deletions(-)

diff --git a/packages/initscripts/initscripts-1.0/bootmisc.sh b/packages/initscripts/initscripts-1.0/bootmisc.sh
index 2a40e0c..37acea5 100755
--- a/packages/initscripts/initscripts-1.0/bootmisc.sh
+++ b/packages/initscripts/initscripts-1.0/bootmisc.sh
@@ -9,7 +9,9 @@
 #
 if test "$DELAYLOGIN" = yes
 then
-  echo "System bootup in progress - please wait" > /etc/nologin
+### Should not be done here but in another initscript as this messes with
+### bootsplashes and just filsl the console with junk
+#  echo "System bootup in progress - please wait" > /etc/nologin
   cp /etc/nologin /etc/nologin.boot
 fi
 
@@ -39,12 +41,13 @@ fi
 #
 # Update /etc/motd.
 #
-if test "$EDITMOTD" != no
-then
-	uname -a > /etc/motd.tmp
-	sed 1d /etc/motd >> /etc/motd.tmp
-	mv /etc/motd.tmp /etc/motd
-fi
+### useless as the motd is empty anyway
+#if test "$EDITMOTD" != no
+#then
+#	uname -a > /etc/motd.tmp
+#	sed 1d /etc/motd >> /etc/motd.tmp
+#	mv /etc/motd.tmp /etc/motd
+#fi
 
 #
 # This is as good a place as any for a sanity check
@@ -59,7 +62,8 @@ fi
 #
 # Update dynamic library cache
 #
-/sbin/ldconfig
+### useless as the package manager should have done this and slows boot a lot
+#/sbin/ldconfig
 
 # Set the system clock from hardware clock
 # If the timestamp is 1 day or more recent than the current time,
diff --git a/packages/initscripts/initscripts_1.0.bb b/packages/initscripts/initscripts_1.0.bb
index b1c4317..8daefc5 100644
--- a/packages/initscripts/initscripts_1.0.bb
+++ b/packages/initscripts/initscripts_1.0.bb
@@ -4,7 +4,7 @@ PRIORITY = "required"
 DEPENDS = "makedevs"
 RDEPENDS = "makedevs"
 LICENSE = "GPL"
-PR = "r109"
+PR = "r110"
 
 SRC_URI = "file://functions \
            file://halt \
-- 
1.5.4.3

